### DISCUSSION TITLE
Leftjoin example on a table

### DISCUSSION THREAD

**Discussion by msywulak** [sadmann7/tablecn#350] [General]
Hello, I've been trying to get a leftjoin on the table to work but have been struggling. It typically comes back to a type issue with nest types not working but I was wondering if someone had an example of a leftjoined table or maybe just a couple hints on the path to go. 

I looked at some of the examples from skateshop with leftjoin, https://github.com/sadmann7/skateshop/blob/main/src/components/tables/products-table.tsx, but they seem to be commented out right now, so I wonder if there's just a general issue that I'm missing. 

Any help is appreciated, thanks!

**Comment by sadmann7** [comment]
hm i changed up the schema a bit, and haven't got much time to work on skateshop.

are you using drizzle? you can follow this if you do: [drizzle leftjoin](https://orm.drizzle.team/docs/joins#:~:text=%7D%5B%5D%3B-,Partial%20select,-If%20you%20need)

**Reply by msywulak** [reply]
It's not so much the drizzle part, it's the downstream implications on the data object that gets created after the join. 

```
const { data, total } = await db.transaction(async (tx) => {
      const data = await tx
        .select()
        .from(logs)
        .limit(per_page)
        .offset(offset)
        .where(where)
        .leftJoin(
           apps,
           eq(logs.applicationId, apps.applicationId),
        )
        .orderBy(
          column && column in logs
            ? order === "asc"
              ? asc(logs[column])
              : desc(logs[column])
            : desc(logs.id),
        );
      const total = await tx
        .select({ count: count() })
        .from(logs)
        .where(where)
        .execute()
        .then((res) => res[0]?.count ?? 0);

      return { data, total };
```

This brings back what you'd expect and causes the data object to throw an error in the `useDataTable` hook within the table component. I tried to create some additional types to account for the new object based on the leftjoin that seem to work on the table and columns components but then throws an error on the `page.tsx` promise. 

It's probably a combination of being newer with types that I can't finesse it appropriately so wanted to see if it was encountered by anyone else and had some recommendations. 

Maybe I'll just create a fork of this repo and add another table to see if I can recreate it there or if I'm just messing up on this internal project.

**Reply by msywulak** [reply]
Yeah seems like it's just a type issue with the return type of data that I need to play with more to get it working. 

```
 const data = await tx
        .select()
        .from(tasks)
        .limit(per_page)
        .offset(offset)
        .where(where)
        .leftJoin(features, eq(tasks.id, features.taskId))
        .orderBy(
          column && column in tasks
            ? order === "asc"
              ? asc(tasks[column])
              : desc(tasks[column])
            : desc(tasks.id)
        )
```

This causes this snippet to have a type error on the data object because it's just expect the Task type so probably just need to keep playing with types to get it to work.
```
  const { table } = useDataTable({
    data,
    columns,
    pageCount,
    // optional props
    filterFields,
    enableAdvancedFilter: featureFlags.includes("advancedFilter"),
    defaultPerPage: 10,
    defaultSort: "createdAt.desc",
  })
```

**Comment by khrnchn** [comment]
working on similar stuff right now, my aim is to pass the data i need from other related tables into the update sheet.
